tensorflow/contrib/rnn/python/ops/rnn_cell.py

```diff
@@ -661,6 +663,109 @@ def _make_tf_features(self, input_feat):
       freq_inputs.append(cur_input)
     return freq_inputs
 
+class ESNCell(rnn_cell.RNNCell):
+  """Echo State Network Cell.
+
+    Based on http://www.faculty.jacobs-university.de/hjaeger/pubs/EchoStatesTechRep.pdf
+    It doesn't consider output feedback, specifically it models just the reservoir.
+
+    Here a practical guide to use Echo State Networks:
+    http://minds.jacobs-university.de/sites/default/files/uploads/papers/PracticalESN.pdf
+
+    Since at the moment TF doesn't provide a way to compute spectral radius
+    of a matrix the echo state property necessary condition `max(eig(W)) < 1` is approximated
+    scaling the norm 2 of the reservoir matrix which is an upper bound of the spectral radius.
+    See https://en.wikipedia.org/wiki/Matrix_norm#Induced_norm
+
+  """
+
+  def __init__(self, num_units, wr2_scale=0.7, connectivity=0.1, leaky=1.0, activation=math_ops.tanh,
+               win_init=init_ops.random_normal_initializer(),
+               wr_init=init_ops.random_normal_initializer(),
+               bias_init=init_ops.random_normal_initializer()):
+    """Initialize the Echo State Network Cell.
+
+    Args:
+      num_units: Int or 0-D Int Tensor, the number of units in the reservoir
+      wr2_scale: desired norm2 of reservoir weight matrix.
+        `wr2_scale < 1` is a sufficient condition for echo state property.
+      connectivity: connection probability between two reservoir units
+      leaky: leaky parameter
+      activation: activation function
+      win_init: initializer for input weights
+      wr_init: used to initialize reservoir weights before applying connectivity mask and scaling
+      bias_init: initializer for biases
+    """
+    self._num_units = num_units
+    self._leaky = leaky
+    self._activation = activation
+
+    def _wr_initializer(shape, dtype):
+      wr = wr_init(shape, dtype=dtype)
+
+      connectivity_mask = \
```

Comment:
backslashes aren't accepted in TF python code; can you find another way to perform the multi-line?  e.g.:

```
connectivity_mask = math_ops.cast(
  ...
  ...)
```